src/marten/middleware/method_override.cr

```diff
@@ -0,0 +1,42 @@
+module Marten
+  abstract class Middleware
+    # Overrides the HTTP method of a request.
+    #
+    # This middleware overrides the HTTP method of a incoming request that include a `_method` field
+    # (in form data or URL-encoded data) or a `X-Http-Method-Override` header,
```

Comment:
The comment mentions that the `X-Http-Method-Override` header is supported but the implementation does not seem to check for this header. Afaics we are only checking for the override param. @treagod we can either add the support for the header later on or as part of this PR, both work fine for me. 🙂